File: backend/backend/src/main/java/com/moovie/backend/domain/usecase/movies/SearchMovies.java
package com.moovie.backend.domain.usecase.movies;

import com.moovie.backend.domain.repository.MoviesRepository;
import com.moovie.backend.domain.model.MovieListing;
import org.springframework.stereotype.Service;

@Service
public class SearchMovies {

    private final MoviesRepository moviesRepository;

    public SearchMovies(MoviesRepository moviesRepository) {
        this.moviesRepository = moviesRepository;
    }

    public MovieListing invoke(String query, int page) {
        return moviesRepository.searchMovies(query, page);
    }
}

File: src/main/kotlin/org/mobyle/domain/usecase/movies/SearchMovies.kt
package org.mobyle.domain.usecase.movies

import kotlinx.coroutines.runBlocking
import org.mobyle.domain.repository.MoviesRepository
import org.mobyle.model.MovieListing

class SearchMovies(private val repository: MoviesRepository) {
    operator fun invoke(query: String, page: Int): MovieListing = runBlocking {
        repository.searchMovies(query, page)
    }
}
